packager/debugger.html

```diff
@@ -58,36 +63,41 @@
       sendReply(JSON.stringify(returnValue));
     }
   }
-}
+};
 
-var ws = new WebSocket('ws://' + window.location.host + '/debugger-proxy');
+function connectToDebuggerProxy() {
+  var ws = new DebuggerWebSocket('ws://' + window.location.host + '/debugger-proxy');
 
-ws.onopen = function() {
-  if (sessionID) {
-    setStatus('Debugger session #' + sessionID + ' active');
-    ws.send(JSON.stringify({replyID: parseInt(sessionID, 10)}));
-  } else {
-    setStatus('Waiting, press ⌘R in simulator to reload and connect');
-  }
-}
+  ws.onopen = function() {
+    if (sessionID) {
+      setStatus('Debugger session #' + sessionID + ' active');
+      ws.send(JSON.stringify({replyID: parseInt(sessionID, 10)}));
+    } else {
+      setStatus('Waiting, press ⌘R in simulator to reload and connect');
+    }
+  };
 
-ws.onmessage = function(message) {
-  var object = JSON.parse(message.data);
-  var sendReply = function(result) {
-    ws.send(JSON.stringify({replyID: object.id, result: result}));
-  }
-  var handler = messageHandlers[object.method];
-  if (handler) {
-    handler(object, sendReply);
-  } else {
-    console.warn('Unknown method: ' + object.method);
-  }
-}
+  ws.onmessage = function(message) {
+    var object = JSON.parse(message.data);
+    var sendReply = function(result) {
+      ws.send(JSON.stringify({replyID: object.id, result: result}));
+    };
+    var handler = messageHandlers[object.method];
+    if (handler) {
+      handler(object, sendReply);
+    } else {
+      console.warn('Unknown method: ' + object.method);
+    }
+  };
 
-ws.onclose = function() {
-  setStatus('Disconnected from proxy. Is node server running?');
+  ws.onclose = function() {
+    setStatus('Disconnected from proxy. Attempting reconnection. Is node server running?');
```

Comment:
> we should probably clear sessionID here, because after disconnecting it will become invalid.

Done. This removes the warning for closing a tab on a closed session too.